php-frontend/src/main/java/org/sonar/php/parser/PHPGrammar.java

```diff
@@ -636,6 +647,28 @@ public DeclaredTypeTree DECLARED_TYPE() {
       b.firstOf(DNF_TYPE(), UNION_TYPE(), INTERSECTION_TYPE(), TYPE()));
   }
 
+  public PropertyHookListTree PROPERTY_HOOK_LIST() {
+    return b.<PropertyHookListTree>nonterminal(PHPLexicalGrammar.PROPERTY_HOOK_LIST).is(
+      f.propertyHookList(
+        b.token(LCURLYBRACE),
+        b.oneOrMore(PROPERTY_HOOK()),
+        b.token(RCURLYBRACE)));
+  }
+
+  public PropertyHookTree PROPERTY_HOOK() {
+    return b.<PropertyHookTree>nonterminal(PHPLexicalGrammar.PROPERTY_HOOK).is(
+      f.propertyHook(
+        b.zeroOrMore(ATTRIBUTE_GROUP()),
+        b.zeroOrMore(MEMBER_MODIFIER()),
+        b.optional(b.token(PHPPunctuator.AMPERSAND)),
+        NAME_IDENTIFIER_OR_KEYWORD(),
+        b.optional(PARAMETER_LIST()),
+        b.optional(b.token(DOUBLEARROW)),
+        b.firstOf(
+          EOS(),
+          INNER_STATEMENT())));
```

Comment:
That's a good idea, makes it also possible to test the behavior better.
Still I can't change body from `Tree`to `StatementTree`, as it can also be just `EOS`which is a Syntax Token.
Example: `int $a { get; }`